ciphers/xor_cipher.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file xor_cipher.cpp
+ * @brief Implementation of [XOR cipher](https://en.wikipedia.org/wiki/XOR_cipher) algorithm.
+ *
+ * @details
+ * In cryptography, the simple XOR cipher is a type of additive cipher, an encryption 
+ * algorithm that operates according to the principles: 
+ *
+ * \f[A {\oplus} 0 = A\f]
+ * \f[A {\oplus} A = 0\f]
+ * \f[ (A {\oplus} B) {\oplus} C = A {\oplus} (B {\oplus} C)\f]
+ * \f[ (B {\oplus} A) {\oplus} B = B {\oplus} 0 = B \f]
+ * 
+ * 
+ * where {\displaystyle \oplus }\oplus  denotes the exclusive disjunction (XOR) operation.
+ * This operation is sometimes called modulus 2 addition (or subtraction, which is identical).
+ * With this logic, a string of text can be encrypted by applying the bitwise XOR operator to
+ * every character using a given key. To decrypt the output, merely reapplying the XOR function 
+ * with the key will remove the cipher.
+ * 
+ * ### Algorithm
+ * Choose the key for encryption and apply XOR operation to each character of a string. 
+ * Reapplying XOR operation to each character of encrypted string will give original string back.
+ * 
+ * \note This program implements XOR Cipher for string with ASCII characters. 
+ * 
+ * @author [Deep Raval](https://github.com/imdeep2905)
+ */
+#include <iostream>
+#include <string>
+#include <cassert>
+
+/** \namespace ciphers
+ * \brief Algorithms for encryption and decryption
+ */
+namespace ciphers {
+    /** \namespace XOR
+     * \brief Functions for [XOR cipher](https://en.wikipedia.org/wiki/XOR_cipher) algorithm.
+     */
+    namespace XOR {   
+        /**
+         * Encrypt given text using XOR cipher.
+         * @param text text to be encrypted
+         * @param key to be used for encyption
+         * @returns new encrypted text
+         */
+        std::string encrypt (const std::string &text, const int &key) {
+            std::string encrypted_text = ""; // Empty string to store encrypted text
+            for (char c : text) { // Going through each character
```

Comment:
```suggestion
            for (auto &c : text) { // Going through each character
```

Reply:
Well, I am not modifying character in string it self. So I guess ```&``` is not necessary ?

Reply:
its ok either way. most compilers would be optimizing that anyways. I suggested because this way, c will be a reference to the const character rather than creating a copy of the value. 
again, not a mandatory change

Reply:
> I suggested because this way, c will be a reference to the const character rather than creating a copy of the value.

I didn't know that.
 New knowledge gained ✔️